Fix subscription callbacks when unsubscribe in AuthorizeService.js in React template

hermanho opened the pull request:
Fix subscription _callbacks array is re-assign when unsubscribe in AuthorizeService.js in React template.

It will throw error after the first unsubscribe is called because the variable _callback is changed to the deleted elements. An array containing the deleted elements is returned when calling splice function in JavaScript means the variable reference should not be re-assigned.

javiercn commented:
@hermanho thanks for the PR.

Could you rebase it on top of the `release/3.1` branch so that we can take it on the next release?

hermanho committed:
Fix subscription callbacks when unsubscribe

Fix subscription _callbacks array is re-assign when unsubscribe in AuthorizeService.js in React template

hermanho commented:
@javiercn It is changed.

javiercn reviewed (approved):
Thanks for the PR!
